### PR TITLE
Add rpm list api

### PR DESCRIPTION
**Pull request by jlsherrill**
This adds a new function to list rpms in a repository version, also refactors existing functions to use query arguments.

### CODE REVIEW

File: pkg/tangy/queries.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"strings"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/exp/rand"
+)
+
+func contentIdsInVersion(repoId string, versionNum int, namedArgs *pgx.NamedArgs) string {
+	ran := rand.Int()
+	repoIdName := fmt.Sprintf("%v%v", "repoName", ran)
+	versionNumName := fmt.Sprintf("%v%v", "versionNum", ran)
```
Comment (rverdile) [inline pkg/tangy/queries.go:19]: can you put a comment above the function saying what the function is supposed to do?

Comment (rverdile) [reply pkg/tangy/queries.go:19]: can you also put a clarifying one for why randomly generating a repoName and versionName is needed?

File: pkg/tangy/rpm.go [hunk 2]
```diff
@@ -183,58 +216,87 @@ func (t *tangyImpl) RpmRepositoryVersionEnvironmentSearch(ctx context.Context, h
...
-    		    WHERE crv.repository_id = '%v' AND crv.number <= %v AND NOT (crv2.number <= %v AND crv2.number IS NOT NULL)
-				AND rp.name ILIKE CONCAT( '%%', '%v'::text, '%%')
-            )
-		`, id, ver, ver, search)
-
-		if i == len(repositoryIDs)-1 {
-			query += fmt.Sprintf(") ORDER BY rp.name ASC LIMIT %v;", limit)
-			break
-		}
+// RpmRepositoryVersionPackageSearch search for RPMs, by name, associated to repository hrefs, returning an amount up to limit
```
Comment (rverdile) [inline pkg/tangy/rpm.go]: Can you update this comment to explain the list function?

**Review by rverdile — COMMENTED** [review]
can you add the list function to the examples the README?

File: internal/test/integration/rpm_test.go [hunk 3]
```diff
@@ -244,6 +244,53 @@ func (r *RpmSuite) TestRpmRepositoryVersionEnvironmentSearch() {
 	assert.Len(r.T(), search, 0)
 }
 
+func (r *RpmSuite) TestRpmRepositoryVersionPackageListNameFilter() {
```
Comment (rverdile) [inline internal/test/integration/rpm_test.go:247]: can you add tests that cover the limit and offset options as well?

**Comment by jlsherrill** [comment]
I think i got all your feedback addressed.

**Review by rverdile — APPROVED** [review]
Looks good! Tested all the methods.

**Pull request merged**